src/fauxtoshop.cpp

```diff
@@ -309,10 +271,10 @@ void getRandomPixelColor(Grid<int> &original
         //The negative and positive radius value forms the upper and lower bounds
         int randomRow = row + randomInteger(-scatterRadius, scatterRadius);
         int randomCol = col + randomInteger(-scatterRadius, scatterRadius);
-            //If the location of the random pixel is in bounds of original image grid
-            //assign the colour of the random pixel to the original pixel
-            //If the condition is not met, keep searching for an in bounds random pixel
-            if (original.inBounds(randomRow, randomCol)){
+        //If the location of the random pixel is in bounds of original image grid
+        //assign the colour of the random pixel to the original pixel
+        //If the condition is not met, keep searching for an in bounds random pixel
+        if (original.inBounds(randomRow, randomCol)){
             original[row][col] = original[randomRow][randomCol];

```

Comment:
If we modify the original grid in place while we're in the process of looping over its pixels, we might end up choosing newly modified pixels as our scatter pixels instead of the original pixels. What we should do is pass in another grid into this function to represent the 'result', and set the result pixels to the scatter pixels from the original grid. E.g.
`result[row][col] = original[randomRow][randomCol]`

Reply:
Aarrrggghhhh rookie mistake 🤦‍♀️

Reply:
This is fixed now. Could you please review getRandomPixelColor() to double check?

Reply:
It looks good now. Nicely done!